client/client.go

```diff
@@ -295,6 +320,33 @@ func newHttpsClient(conf Config) (*http.Client, error) {
 	transport := http.Transport{
 		TLSClientConfig: &tlsc,
 		Proxy:           http.ProxyFromEnvironment,
+		DialTLS: func(network string, addr string) (net.Conn, error) {
+			//contextSSL := ctx.Value("ssl").(*openssl.Ctx)
+			contextSSL, err := openssl.NewCtx() // probably should consider reusing the context, but we
+			// have to propagate it with request, to get it here see^
+			err = contextSSL.LoadVerifyLocations("./client/" + conf.ServerCert, "")
+			if err != nil {
+				return nil, err
+			}
+			flags := openssl.DialFlags(0)
+
+			if conf.NoVerify {
+				flags = openssl.InsecureSkipHostVerification
+			}
+
+			conn, err := openssl.Dial("tcp", addr, contextSSL, flags)
+			v := conn.VerifyResult()
+			if conn == nil || err != nil {
+				return nil, err
+			}
+			if v != openssl.Ok {
+				if v == openssl.CertHasExpired || v == openssl.DepthZeroSelfSignedCert {
+					return nil, errors.New(fmt.Sprintf("certificate has expired, openssl says: %d", v))
```

Comment:
`errors.Errorf` or `fmt.Errorf`?

Reply:
sure Alf, thanks. (it will go into the final PR)